Firmware/cmdqueue.cpp

```diff
@@ -251,6 +252,19 @@ void cmdqueue_dump_to_serial()
 static const char bufferFull[] PROGMEM = "\" failed: Buffer full!";
 static const char enqueingFront[] PROGMEM = "Enqueing to the front: \"";
 
+
+void enquecommandf(const char *fmt, ...)
+{
+    // MAX_CMD_SIZE is 96, but for formatting
+    // string we usually don't need more than 30 bytes
+    char cmd_buffer[30];
```

Comment:
Idea: Could we maybe work around the need for allocating an array on the stack? I think it would be more efficient if we made a function in the cmdqueue code to get a pointer for writing (so it allocates a place in the cmdqueue that can fit MAX_CMD_SIZE, and then we format with vsnprintf_P on that pointer and then we "finalize" the command by telling the cmdqueue what is the length of the string that was added, so that less than MAX_CMD_SIZE is reserved after doing this enqueuing operation. This way you don't allocate the buffer on the stack (less stack footprint, so less chaces of overwriting static variables) and it also avoids having to copy data from one temporary buffer to the cmdqueue, so it should be faster (marginally, but still faster nonetheless).

Reply:
This is actually a good idea, because the formatted string ends up in the cmd queue anyway, so why not writing directly into the target memory area ... might not be simple to achieve in the current API though...

Reply:
The tricky part here would be knowing the formatted string in advance to prepare the buffer's internal indices correctly.
The funny point is, that `enquecommand` actually may silently skip a g-code line entirely when `cmdqueue_could_enqueue_back` fails to find enough space in the buffer.